src/cmd/cos_customizer/install_gpu.go

```diff
@@ -127,11 +130,18 @@ func (i *InstallGPU) validate(ctx context.Context, gcsClient *storage.Client, fi
 	if i.NvidiaDriverVersion == "" {
 		return fmt.Errorf("version must be set")
 	}
-	gpuAlreadyConf, err := fs.StateFileContains(files.StateFile, fs.Builtin, gpuScript)
+	stateFileAlreadyConf, err := fs.StateFileContains(files.StateFile, fs.Builtin, gpuScript)
 	if err != nil {
 		return err
 	}
-	if gpuAlreadyConf {
+	var provConfigAlreadyConf bool
+	for _, s := range provConfig.Steps {
+		if s.Type == "InstallGPU" {
+			provConfigAlreadyConf = true
+			break
+		}
+	}
+	if stateFileAlreadyConf || provConfigAlreadyConf {
```

Comment:
is the intent here something like:
`gpuAlreadyConf := stateFileAlreadyConf || provConfigAlreadyConf`?

Reply:
Yeah, that's the intent. My goal is to have validation for both the state file and the provisioner config, and both need to validate that the GPU step is only invoked once.